Give more correct descriptions of negated And conditions

agrrr3 opened the pull request:
The ::description() of FOCS And conditions gave the same output no matter if the condition is negated or not.

This PR handles uses de morgans law for transformation in the Condition.cpp, describing NOT AND as an OR and pushing the negation to the operands. Same goes for NOT OR which gets transformed to AND.

```
Not And [  DesignHasHull name ="SH_COLONY_BASE"     ( PartsInShipDesign design = LocalCandidate.DesignID < 1 ) ]
(either not that contains the hull Colony Base Hull, or not when PartsInShipDesign < 1 )
```

Also NOT NOT nullifies itself now by negation switching.

A previous commit used de morgans law directly on the stringtable level, which allowed to use 'neither'/'nor' in english, but as it prevented negative operands to cancel the negation out i found that to be less clear in the end. 

Also note that one could pull up AND conditions inside of AND conditions and OR conditions inside of OR condition, making the description more clear. But probably in those cases one should fix the condition instead of the description, so its actually good to keep it smelly like it is.

Commit:
Give more correct descriptions of negated And conditions

Dilvish-fo reviewed (commented):
I think that this set of changes would accomplish my proposed adjustment

Commit:
Fix descriptions: not-not, and, not-and, or, not-or

* maybe not-and and not-or should actually push negation down to parts

Commit:
Describe NOT AND (A B..) as OR (NOT A NOT B..) by relying on negated description of A, B.. (inverse for NOR OR)

* Describe NOT AND (A B..) as  OR (NOT A NOT B..) by relying on negated description of A, B..
* Describe NOT  OR (A B..) as AND (NOT A NOT B..) by relying on negated description of A, B..
* there are stringtable entries for the NOT_AND, NOT_OR as maybe we want to be able to see a difference between a NOT AND and a OR.
  Also it could be that there is a wording difference in languages different from english. Else this could be removed.

Alternative implementation of DeMorgan via stringtables can be found in previous commit a4b44327b780d084d7721dc9abf3f6e0f38fc9a5

Reasoning:
I found its worthwhile to actually push the negation through to the operands because that gives the possibility for more helpful messages as double negation has a chance to cancel itself out (instead of '(neither an object that is not a ORB_SY...' you get a '(an object that is a ORB_SY and ..)'. Also in english this is really fine.
